src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_map.c

```diff
@@ -1,108 +1,918 @@
+
+                while (true) {
+                    pauseCtx->cursorPoint[PAUSE_WORLD_MAP]--;
+                    if (pauseCtx->cursorPoint[PAUSE_WORLD_MAP] < 0) {
+                        KaleidoScope_MoveCursorToSpecialPos(play, PAUSE_CURSOR_PAGE_LEFT);
+                        pauseCtx->cursorItem[PAUSE_MAP] = PAUSE_ITEM_NONE;
+                        break;
+                    }
+                    if (pauseCtx->worldMapPoints[pauseCtx->cursorPoint[PAUSE_WORLD_MAP]]) {
+                        break;
+                    }
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_kaleido_scope/func_8081D240.s")
+                if (pauseCtx->cursorSpecialPos == 0) {
+                    pauseCtx->cursorItem[PAUSE_MAP] = pauseCtx->cursorPoint[PAUSE_WORLD_MAP];
+                    // Used as cursor vtxIndex
+                    pauseCtx->cursorSlot[PAUSE_MAP] = 31 + pauseCtx->cursorPoint[PAUSE_WORLD_MAP];
+                }
+                play_sound(NA_SE_SY_CURSOR);
+                sStickAdjTimer = 0;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_kaleido_scope/func_8081D6DC.s")
+        if (!pauseCtx->worldMapPoints[pauseCtx->cursorPoint[PAUSE_WORLD_MAP]]) {
+            pauseCtx->cursorItem[PAUSE_MAP] = PAUSE_ITEM_NONE;
+        }
+        if (oldCursorPoint != pauseCtx->cursorPoint[PAUSE_WORLD_MAP]) {
+            play_sound(NA_SE_SY_CURSOR);
+        }
+    } else if (pauseCtx->state == 0x17) {
+        pauseCtx->cursorColorSet = 4;
+        oldCursorPoint = pauseCtx->cursorPoint[PAUSE_WORLD_MAP];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_kaleido_scope/func_8081E118.s")
+        if (pauseCtx->stickAdjX > 30) {
+            pauseCtx->unk_298 = 4.0f;
+            sStickAdjTimer = 0;
+            do {
+                pauseCtx->cursorPoint[PAUSE_WORLD_MAP]++;
+                if (pauseCtx->cursorPoint[PAUSE_WORLD_MAP] > OWL_WARP_STONE_TOWER) {
+                    pauseCtx->cursorPoint[PAUSE_WORLD_MAP] = OWL_WARP_GREAT_BAY_COAST;
+                }
+            } while (!pauseCtx->worldMapPoints[pauseCtx->cursorPoint[PAUSE_WORLD_MAP]]);
+        } else if (pauseCtx->stickAdjX < -30) {
+            pauseCtx->unk_298 = 4.0f;
+            sStickAdjTimer = 0;
+            do {
+                pauseCtx->cursorPoint[PAUSE_WORLD_MAP]--;
+                if (pauseCtx->cursorPoint[PAUSE_WORLD_MAP] < OWL_WARP_GREAT_BAY_COAST) {
+                    pauseCtx->cursorPoint[PAUSE_WORLD_MAP] = OWL_WARP_STONE_TOWER;
+                }
+            } while (!pauseCtx->worldMapPoints[pauseCtx->cursorPoint[PAUSE_WORLD_MAP]]);
+        } else {
+            sStickAdjTimer++;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_kaleido_scope/func_8081E7D8.s")
+        //! TODO: Is the `0xA4` here related to `0xA3` being the last recored item in the `ItemId` enum?
+        pauseCtx->cursorItem[PAUSE_MAP] = sOwlWarpPauseItems[pauseCtx->cursorPoint[PAUSE_WORLD_MAP]] - 0xA4;
```

Comment:
Really this comment goes with the array, right?

Reply:
Maybe both, but I think the note should stay here as the value being offset by `0xA4` seems important for being only 1 value above 0xA3

Reply:
Maybe just make a temporary enum and put the note there, since it's clear that this 0xA4 is the same as the one in that array.

Reply:
I had it like this at one point, do you think it's better like this?
```c
u16 sOwlStatuePauseItems[] = {
    0xB + 0xA4, // OWL_STATUE_GREAT_BAY_COAST
    0xF + 0xA4, // OWL_STATUE_ZORA_CAPE
    0x6 + 0xA4, // OWL_STATUE_SNOWHEAD
    0xD + 0xA4, // OWL_STATUE_MOUNTAIN_VILLAGE
    0x5 + 0xA4, // OWL_STATUE_CLOCK_TOWN
    0xE + 0xA4, // OWL_STATUE_MILK_ROAD
    0x4 + 0xA4, // OWL_STATUE_WOODFALL
    0xC + 0xA4, // OWL_STATUE_SOUTHERN_SWAMP
    0x8 + 0xA4, // OWL_STATUE_IKANA_CANYON
    0xA + 0xA4, // OWL_STATUE_STONE_TOWER
};
```

Reply:
I don't like it much, not sure there's a better way for the time being though.

Reply:
I think stick with current for now, sort it out later.